### PR TITLE
Resolve #1207, #1221, #1224, #1223, #1225, #1222, #1196

### LINKED ISSUE #1207: Btc sync should store balance in string

### PR DESCRIPTION
**Pull request by timothywangdev**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -16,11 +16,12 @@ REACT_APP_ETHEREUM_NETWORK=rinkeby
 REACT_APP_INFURA_API_KEY=9775236247814824bc231e65b1d4972a
 REACT_APP_BTC_NETWORK=testnet
 REACT_APP_DAI_ADDRESS=0x4aacB7f0bA0A5CfF9A8a5e8C0F24626Ee9FDA4a6
-REACT_APP_CHAINSFER_API_ENDPOINT=https://3zowl9pdcl.execute-api.us-east-1.amazonaws.com/Prod
```
Comment (InVincible2016) [inline .env]: Is this intentional?

File: src/accounts/BitcoinAccount.js [hunk 2]
```diff
@@ -91,19 +93,19 @@ export default class BitcoinAccount implements IAccount<AccountData> {
...
-            walletType: accountData.walletType,
-            xpub: accountData.xpub
-          }))
-        : (_accountData.id = JSON.stringify({
-            cryptoType: accountData.cryptoType,
-            walletType: accountData.walletType,
-            address: accountData.address
-          }))
-    }
+    _accountData.hdWalletVariables.xpub
```
Comment (InVincible2016) [inline src/accounts/BitcoinAccount.js]: should probably also make changes to EthereumAccount.js?

File: src/actions/transferActions.js [hunk 3]
```diff
@@ -572,6 +573,301 @@ async function _getTransfer (transferId: ?string, receivingId: ?string) {
...
+      // 2. gather internal tx (sent from a contract)
+      response = await axios.get(
+        `${url.ETHERSCAN_API_URL}/api?module=account&action=txlistinternal&address=${account.address}&page=1&offset=${MAX_TXS}&sort=desc&apikey=${env.REACT_APP_ETHERSCAN_API_KEY}`
+      )
+      const txsInternal = response.data.result
+
+      // combine normal txs and internal txs, order in non-increasing timestamp
+      txs = [...txsNormal, ...txsInternal].sort((txA, txB) => txB - txA)
+    }
+  }
```
Comment (InVincible2016) [inline src/actions/transferActions.js]: Put line 628 to 667 into a separate function.

File: src/actions/transferActions.js [hunk 4]
```diff
@@ -572,6 +573,301 @@ async function _getTransfer (transferId: ?string, receivingId: ?string) {
...
+
+  // tx always have property hash regardless of cryptoType
```
Comment (InVincible2016) [inline src/actions/transferActions.js]: ok...but not sure what this comment is for??

File: src/actions/transferActions.js [hunk 5]
```diff
@@ -572,6 +573,301 @@ async function _getTransfer (transferId: ?string, receivingId: ?string) {
...
+          rv[txHash].standaloneTx = {
+            from,
+            to,
+            value,
+            cryptoType: account.cryptoType
+          }
+        }
+      }
+    }
+  }
```
Comment (InVincible2016) [inline src/actions/transferActions.js]: Put line 749 - 842 input a separate function

File: src/actions/transferActions.js [hunk 6]
```diff
@@ -572,6 +573,301 @@ async function _getTransfer (transferId: ?string, receivingId: ?string) {
...
+      console.log(sendTxHash, txHash2Idx[sendTxHash])
+      // this should not happen
+      throw new Error('transferData does not have matched txHash')
+    }
+
+    // fill in data
+    transferData = unmarshalAccount(transferData)
+    rv[txs[idx].hash] = { timestamp }
+    rv[txs[idx].hash].transferData = { ...transferData, transferType }
+  }
```
Comment (InVincible2016) [inline src/actions/transferActions.js]: put 718 - 746 into a separate function.

File: .env (hunk 1 above)
Comment (timothywangdev) [reply .env]: no

File: src/accounts/BitcoinAccount.js (hunk 2 above)
Comment (timothywangdev) [reply src/accounts/BitcoinAccount.js]: already made

File: src/actions/transferActions.js (hunk 3 above)
Comment (timothywangdev) [reply src/actions/transferActions.js]: done

File: src/actions/transferActions.js (hunk 4 above)
Comment (timothywangdev) [reply src/actions/transferActions.js]: moved

File: src/actions/transferActions.js (hunk 5 above)
Comment (timothywangdev) [reply src/actions/transferActions.js]: cannot move due to multiple dependencies

File: src/actions/transferActions.js (hunk 6 above)
Comment (timothywangdev) [reply src/actions/transferActions.js]: cannot move due to multiple dependencies

Comment (timothywangdev) [reply src/actions/transferActions.js]: cannot move due to multiple dependencies

**Pull request merged**